demos/TicTacToe/Classes/TicTacToeScene.cpp

```diff
@@ -0,0 +1,81 @@
+#include "TicTacToeScene.h"
+
+#include <array>
+#include <cstdlib>
+#include <iostream>
+
+#include "cocos2d.h"
+
+USING_NS_CC;
+
+const int kNumberOfTiles = 9;
+const int kMaxMovesPerPlayer = 1 + kNumberOfTiles / 2;
+const int kTilesX = std::sqrt(kNumberOfTiles);
+const int kTilesY = std::sqrt(kNumberOfTiles);
+const double kScreenWidth = 600;
+const double kScreenHeight = 600;
+const double kTileWidth = (kScreenWidth / kTilesX);
+const double kTileHeight = (kScreenHeight / kTilesY);
+const int kNumberOfPlayers = 2;
+std::array<std::string, kNumberOfPlayers> kMoveImageFileNames = {
+    "tic_tac_toe_x.png", "tic_tac_toe_o.png"};
+
+Scene* TicTacToe::createScene() {
+  auto scene = Scene::create();
+  auto layer = TicTacToe::create();
+
+  scene->addChild(layer);
+
+  return scene;
+}
+
+bool TicTacToe::init() {
+  if (!Layer::init()) {
+    return false;
+  }
+
+  auto file_names_it = std::begin(kMoveImageFileNames);
+
+  // TODO(grantpostma): This should reflect the size that is set in AppDelegate.
+  // Should modify kTileWidth and kTileHeight based on that size auto
+  // kScreenWidth = Director::getInstance()->getWinSize().width; auto
+  // kScreenHeight =  Director::getInstance()->getWinSize().height;
+
+  auto board_sprite = Sprite::create("tic_tac_toe_board.png");
+  board_sprite->setPosition(0, 0);
+  board_sprite->setAnchorPoint(Vec2(0.0, 0.0));
+
+  auto touchListener = EventListenerTouchOneByOne::create();
+  touchListener->onTouchBegan = [=](Touch* touch,
```

Comment:
https://google.github.io/styleguide/cppguide.html#Lambda_expressions

> Prefer explicit captures when the lambda will escape the current scope.

Only capture the variables you need in the lambda.